src/DataProvider/StateLegacyDataProvider.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataProvider;
+
+use ApiPlatform\State\ProviderInterface;
+
+/**
+ * State provider legacy bridge to use new Providers with pre-2.7 resources.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ * @experimental
+ */
+class StateLegacyDataProvider implements DenormalizedIdentifiersAwareItemDataProviderInterface, ContextAwareCollectionDataProviderInterface
+{
+    public function __construct(private ?ProviderInterface $provider = null)
```

Comment:
Why using PHP 8 only syntax?

Reply:
because this class is meant to be used with php8 ? :D 

I really can change this but I'm not sure there is an interest in doing so.

Reply:
The StateLegacyDataProvider is used by the ChainDataProvider which can be used in PHP 7 (IMHO), so I guess PHP 8 code should not be compatible here